docs/index.asciidoc

```diff
@@ -0,0 +1,289 @@
+
+include::{include_path}/plugin_header.asciidoc[]
+
+==== Description
+
+Using this input you can receive events from {esf-name} over http(s) connections to the configured <<plugins-{type}s-{plugin}-port>>.
+
+[id="plugins-{type}s-{plugin}-ext-field"]
+====== Minimum Configuration
+[cols="3a,2a"]
+|=======================================================================================================================
+|SSL Enabled              |SSL Disabled
+
+|
+
+[source,subs="+macros,attributes"]
+----
+input {
+  {plugin} {
+    <<plugins-{type}s-{plugin}-port,port>> => 8080
+    <<plugins-{type}s-{plugin}-ssl_certificate,ssl_certificate>> => "/path/to/logstash.crt"
+    <<plugins-{type}s-{plugin}-ssl_key,ssl_key>> => "/path/to/logstash.key"
+  }
+}
+----
+
+|
+
+[source,subs="+macros,attributes"]
+----
+input {
+  elastic_serverless_forwarder {
+    <<plugins-{type}s-{plugin}-port,port>> => 8080
+    <<plugins-{type}s-{plugin}-ssl,ssl>> => false
+  }
+}
+----
+
+|=======================================================================================================================
+
+[id="plugins-{type}s-{plugin}-enrichment"]
+==== Enrichment
+
+This input provides _minimal enrichment_ on events, and avoids including information about itself, the client from which it received the data, or about the original event as-decoded from the request.
+If the decoded event has a valid ISO8601-encoded `@timestamp`, it will be used. Otherwise this required field will be populated with the current time.
+
+NOTE: Senders are advised to use care with respect to fields that are {logstash-ref}/processing.html#reserved-fields[reserved in Logstash].
+
+[id="plugins-{type}s-{plugin}-blocking"]
+==== Blocking Behavior
+
+When the Logstash pipeline that is receiving events from this input has a full queue, it prevents the pipeline's inputs from putting additional events into the queue until the capacity to do so becomes available.
+This may result in an HTTP request timing out with some or all of its events unprocessed.
+A client that receives an HTTP request timeout is expected to retry the entire request, even though it may cause duplication of data.
+
+When this plugin is blocked, it will reject _new_ requests with HTTP `429 Too Many Requests`.
+Clients that receive `429`-s are expected to wait a moment before retrying the request — exponential back-off is encouraged to ease flood recovery.
+
+[id="plugins-{type}s-{plugin}-security"]
+==== Security
+
+This plugin has SSL on-by-default.
+
+At a minimum, you will need to either configure the plugin to present its identity, or disable SSL.
+
+Additionally, you may wish to authenticate clients using SSL client authentication, and/or authenticate requests using HTTP Basic authentication as described below.
+
+===== SSL Identity
+
+In order to establish SSL with a client, this input plugin will need to present an SSL certificate that the client trusts and have access to the associated key.
```

Comment:
did we get understanding/alignment on how this plays with ESF's ssl fingerprint stuff?

Reply:
_maybe_.

I am a little unclear about how ESF can be configured to establish trust.

Off-thread you mentioned:

> we recently added support for self-signed certificate verification based on fingerprint (https://github.com/elastic/elastic-serverless-forwarder/pull/173). 

 - I assume ESF has access to a list of global root CA's (verisign, etc),
 - I believe it _cannot_ be configured with additional CA's because doing so would require side-loading storage.

The issue linked above seems to indicate that it can be configured with an `ssl_assert_fingerprint`, which implies will be used to validate the specific self-signed certificate presented by the host it connects to.

My question is this:
 - if the server 
   - presents a valid _chain_ of certificates to ESF as documented:
     > This certificate _MUST_ be PEM-formatted, and _MAY_ contain a chain of certificates starting with the certificate that identifies itself, followed by zero or more ordered intermediates optionally ending with the root signing authority.
     > Providing a complete chain allows clients to trust our certificate if their configuration allows them to trust one of our intermediates.
     > -- [`ssl_certificate`](https://github.com/logstash-plugins/logstash-input-elastic_serverless_forwarder/pull/1/files#diff-cae5619b3d18ec99c5ccd0a9f6de0c6d3f53343c64692444551a7d29da6863e7R194-R195),
   - AND the fingerprint of the first-in-chain identity certificate does _NOT_ match,
   - AND the fingerprint of one of the _ancestor_ certificates provided in the chain matches the `ssl_assert_fingerprint`,
   - will it allow the connection?
 - OR does ESF need to know _ALL_ possible fingerprints of _ALL_ downstream hosts in advance?

***

In Logstash we implemented something like this using a flag called `ca_trusted_fingerprint`. Our [implementation](https://github.com/elastic/logstash/blob/v8.3.3/logstash-core/src/main/java/org/logstash/util/CATrustedFingerprintTrustStrategy.java#L60-L85) is a `org.apache.http.conn.ssl.TrustStrategy` (because we are working with Apache's HTTP libraries for Java) to tell our SSL engine that the chain:
 - is trusted (and therefore it does _not_ need further validation against the CA's in our truststore),
 - XOR is not trusted (and therefore it _does_ need validation against the CA's in our truststore),
 - XOR it it is invalid (e.g., cert #2 in chain is trustworthy by the fingerprint, but cert #1 doesn't have a valid signature from it, or it is expired).

One thing to note is that even if our `TrustStrategy` trusts a chain and bypasses the truststore, our client libraries still validate that the trusted certificate belongs to the host we are connecting to (via subject and/or subject alternative names).